include/zephyr/senss/senss_sensor.h

```diff
@@ -0,0 +1,615 @@
+ * API support get separated range for each data field, or global range for
+ * all data fields.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param value The data buffer to receive value.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+typedef int (*senss_sensor_get_range_t)(
+		senss_sensor_handle_t handle,
+		int index, double *value);
+
+/**
+ * @brief Set current sensor data fifo size callback
+ *
+ * Some sensors especially for physical sensors, support fifo, this API can
+ * configure the current fifo size.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param samples The sample number to set for fifo.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+typedef int (*senss_sensor_set_fifo_t)(
+		senss_sensor_handle_t handle,
+		uint32_t samples);
+
+/**
+ * @brief Get current sensor data fifo size
+ *
+ * Some sensors especially for physical sensors, support fifo, this API can
+ * get the current fifo size.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param samples The data buffer to receive the fifo sample number.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+typedef int (*senss_sensor_get_fifo_t)(
+		senss_sensor_handle_t handle,
+		uint32_t *samples);
+
+/**
+ * @brief Set current sensor data offset callback
+ *
+ * Some sensors especially for physical sensors, such as accelerometer senors,
+ * as data drift, need configure offset calibration.
+ *
+ * Since each sensor type may have multiple data fields for it's value, this
+ * API support set separated offset for each data field, or global offset for
+ * all data fields.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param value The offset value to be set.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+typedef int (*senss_sensor_set_offset_t)(
+		senss_sensor_handle_t handle,
+		int index, double value);
+
+/**
+ * @brief Get current sensor data offset callback
+ *
+ * Some sensors especially for physical sensors, such as accelerometer senors,
+ * as data drift, need configure offset calibration.
+ *
+ * Since each sensor type may have multiple data fields for it's value, this
+ * API support get separated offset for each data field, or global offset for
+ * all data fields.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param value The data buffer to receive the offset value.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+typedef int (*senss_sensor_get_offset_t)(
+		senss_sensor_handle_t handle,
+		int index, double *value);
```

Comment:
This seems to replicate logic in sensors v2, what's the reasoning behind needing it?

Reply:
These APIs looks similar because its also for sensors' common functions.  

The `sensor subsystem` provides a sensor framework can write `Physical` and fusion `Virtual` sensors, these APIs are for how to implement them.

For `Physical` sensors,  the implementation is a thin wrapper with current zephyr sensor device API,  or V2 API. 

And why does `sensor subsystem` need define these APIs,  just as we discussed in https://github.com/zephyrproject-rtos/zephyr/pull/45370#issuecomment-1123127132,  and I think the discussion will be go on.  :)